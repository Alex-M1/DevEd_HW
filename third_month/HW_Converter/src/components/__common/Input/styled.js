import styled from 'styled-components';

export const StInput = styled.input`
  border: none;
  border-bottom: 1px solid black;
  background: #e1e1e1;
  outline: none;
`;
