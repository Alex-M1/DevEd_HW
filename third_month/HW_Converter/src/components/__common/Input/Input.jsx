import React from 'react';
import PropTypes from 'prop-types';
import { StInput } from './styled';

function Input({ value, onChange, readOnly = false }) {
  const handleInputChange = (e) => {
    const regExp = /[0-9]/;
    if (!regExp.test(+e.target.value)) return;
    onChange(e.target.value);
  };
  return (
    <StInput value={value} onChange={handleInputChange} readOnly={readOnly} />
  );
}

export default Input;

Input.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func,
  readOnly: PropTypes.bool,
};
