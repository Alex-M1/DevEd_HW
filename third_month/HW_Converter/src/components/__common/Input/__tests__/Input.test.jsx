import { mount, shallow } from 'enzyme';
import React from 'react';
import Input from '../Input';

describe('Form', () => {
  const props = {
    value: '',
    onChange: jest.fn(),
    readOnly: false,
  };
  afterEach(() => {
    props.onChange.mockClear();
  });
  it('Should match snapshot', () => {
    const component = shallow(<Input {...props} />);
    expect(component.html()).toMatchSnapshot();
  });
  it('Should have StInput', () => {
    const component = mount(<Input {...props} />);
    expect(component.find('styled__StInput')).toHaveLength(1);
  });
  it('Should have call onChange', () => {
    const newValue = '123';
    const component = mount(<Input {...props} />);
    component.find('styled__StInput').simulate('change', { target: { value: newValue } });
    expect(component.props().onChange).toHaveBeenCalledTimes(1);
  });
  it('Should have call onChange && value', () => {
    const newValue = '123';
    const component = mount(<Input {...props} />);
    component.find('styled__StInput').simulate('change', { target: { value: newValue } });
    expect(component.props().onChange).toHaveBeenCalledWith(newValue);
  });
  it('Should have call onChange && invalid value', () => {
    const newValue = 'sd';
    const component = mount(<Input {...props} />);
    component.find('styled__StInput').simulate('change', { target: { value: newValue } });
    expect(component.props().onChange).toHaveBeenCalledTimes(0);
  });
});
