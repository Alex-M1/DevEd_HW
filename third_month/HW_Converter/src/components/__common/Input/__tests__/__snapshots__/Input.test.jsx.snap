// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Should match snapshot 1`] = `"<input value=\\"\\" class=\\"styled__StInput-sc-c3z9ep-0 feLlzU\\"/>"`;
